### PR TITLE
Forgot password api

### PR DESCRIPTION
**Pull request by samarv**
# Description

reset password by sending password change URL to user ID

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My code has been reviewed by at least one peer
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: backend/server/routes.js [hunk 1]
```diff
@@ -15,6 +16,8 @@ module.exports = {
       res.send('{"message":"Developer Profiles API"}');
     });
 
+    server.use('/api/saveresethash', forgotPasswordRouter);
```
Comment (d-vail) [inline backend/server/routes.js]: You don't need to tell API users that we are resetting a hash. I would change it to `/api/reset-password`

File: backend/package.json [hunk 2]
```diff
@@ -36,6 +37,8 @@
     "helmet": "^3.13.0",
     "jsonwebtoken": "^8.3.0",
     "mongoose": "^5.2.14",
+    "nodemailer": "^4.6.8",
+    "passport": "^0.4.0",
```
[the remark this answers is not in the document]
Comment (samarv) [reply backend/package.json]: removed passport.

File: backend/package.json [hunk 3]
```diff
@@ -36,6 +37,8 @@
     "helmet": "^3.13.0",
     "jsonwebtoken": "^8.3.0",
     "mongoose": "^5.2.14",
+    "nodemailer": "^4.6.8",
```
[the remark this answers is not in the document]
Comment (samarv) [reply backend/package.json]: added sendgrid

File: backend/api/routes/ForgotPassword.router.js [hunk 4]
```diff
@@ -0,0 +1,108 @@
+/* eslint-disable */
```
[the remark this answers is not in the document]
Comment (samarv) [reply backend/api/routes/ForgotPassword.router.js]: as a team we decide to disable es-lint till the we have the functionality.

**Pull request merged**